server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -283,6 +286,35 @@ public boolean setExpiration(String name, Instant expiration) {
     return Optional.empty();
   }
 
+  /**
+   * @param names of secrets series to look up secrets by.
+   * @return Secrets matching input parameters.
+   */
+  public List<SecretSeriesAndContent> getSecretsByName(List<String> names) {
+    checkArgument(!names.isEmpty());
+
+    SecretContentDAO secretContentDAO = secretContentDAOFactory.using(dslContext.configuration());
+    SecretSeriesDAO secretSeriesDAO = secretSeriesDAOFactory.using(dslContext.configuration());
+
+    List<SecretSeries> multipleSeries = secretSeriesDAO.getMultipleSecretSeriesByName(names);
+
+    List<SecretSeriesAndContent> ret = new ArrayList<SecretSeriesAndContent>();
+
+    for (SecretSeries series : multipleSeries) {
+      if (series.currentVersion().isPresent()) {
+        long secretContentId = series.currentVersion().get();
+        Optional<SecretContent> secretContent =
+                secretContentDAO.getSecretContentById(secretContentId);
+        if (secretContent.isPresent()) {
+          ret.add(SecretSeriesAndContent.of(series, secretContent.get()));
+        } else {
+          throw new NotFoundException("Secret not found.");
```

Comment:
If we have batch endpoint, do we want to fail if even one secret is not found or do we want to just return what's been found? How would an endpoint sending a non-existent secret name know how to drop an offending name and carry on?

Reply:
We've had lengthy discussions around whether we should keep this best-effort and ended on no. A.) It is possible for the client to describe secrets and see which they can access, this only becomes an issue if a secret changes right after a describe B.) the error message informs the client which secret was causing the error. A reason for this is keeping things simple - best-effort would have required being able to respond with different error messages per secret - that would have made the parsing on the client side more complicated.